src/tmpfiles/tmpfiles.c

```diff
@@ -2003,6 +2005,177 @@ static int glob_item_recursively(Item *i, fdaction_t action) {
         return r;
 }
 
+static int rm_if_wrong_type_safe(
+                mode_t mode,
+                int parent_fd,
+                const struct stat *parent_st /* Only used if follow is true. */,
+                const char *name,
+                bool follow) {
+        _cleanup_free_ char *parent_name = NULL;
+        struct stat st;
+        int r;
+        const char *e;
+
+        assert(name);
+        assert((mode & ~S_IFMT) == 0);
+        if (follow)
+                assert(parent_st);
```

Comment:
combine these two lines into:

```c
assert(!follow || parent_st);
```

note that assert() can be removed if `-DNDEBUG` is defined during build. and you want to remove the `if (follow)` check in that case too.

Reply:
Thanks